src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -344,8 +344,10 @@ def translate_symbol_ref(
     else:
         sym_value = str(node.id)
         if sym_value in let_symbols:
-            return [let_symbols[sym_value]]
```

Comment:
Before you were returning when you encountered a let symbol, but now you run the function.
From the let symbols you get the `sym_node`, however, in the `if` bellow you create a new access node.
I just want to make sure that this is intentional.

Reply:
It is intentional, indeed. Before I was assuming that the let symbols would always be used in the same state where they were created, so I could return and use the access node. With branching and state transitions, I cannot assume it so I always create a new access node in the current state. By design, now, the let symbols are always visited (and therefore initialized) in a state that precedes current state.

Reply:
Since it kind of violates the design principles of the SDFG (it is still allowed because simplify will save us) I would add a comment that explains this.